lib/spack/docs/spack_workflows.rst

```diff
@@ -0,0 +1,583 @@
+===============
+Spack Workflows
+===============
+
+The process of using Spack involves building packages, running
+binaries from those packages, and developing software that depends on
+those packages.  For example, one might use Spack to build the
+`netcdf` package, use `spack load` to run the `ncdump` binary, and
+finally, write a small C program to read/write a particular NetCDF file.
+
+Spack supports a variety of workflows to suit a variety of situaions
+and user preferences, there is no single way to do all these things.
+This chapter demonstrates different workflows that have been
+developed, pointing out the pros and cons of them.
+
+###########
+Definitions
+###########
```

Comment:
I don't think `#` is what you want here. If you read [Sphinx's documentation on sections](http://www.sphinx-doc.org/en/stable/rest.html#comments), you'll see that `#` is the highest level. We don't use anything above `=` in our documentation currently. You'll probably want to make this `-` and move the below sections to `^`